src/test/java/seedu/address/model/person/DonorTest.java

```diff
@@ -0,0 +1,84 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_NAME_BOB;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_NRIC_BOB;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_PHONE_BOB;
+import static seedu.address.testutil.TypicalPersons.ALICE;
+import static seedu.address.testutil.TypicalPersons.BOB;
+import static seedu.address.testutil.TypicalPersons.JOHN;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.DonorBuilder;
+import seedu.address.testutil.PersonBuilder;
+
+public class DonorTest {
+
+    @Test
+    public void isSamePerson() {
+        // same object -> returns true
+        assertTrue(JOHN.isSamePerson(JOHN));
+
+        // null -> returns false
+        assertFalse(JOHN.isSamePerson(null));
+
+        Donor editedJohn = new DonorBuilder(JOHN).build();
+
+        // different nric -> returns false
+        editedJohn = new DonorBuilder(JOHN).withNric(VALID_NRIC_BOB).build();
+        assertFalse(JOHN.isSamePerson(editedJohn));
+
+        // different phone -> returns true
+        editedJohn = new DonorBuilder(JOHN).withPhone(VALID_PHONE_BOB).build();
+        assertTrue(JOHN.isSamePerson(editedJohn));
+
+        // different name -> returns true
+        editedJohn = new DonorBuilder(JOHN).withName(VALID_NAME_BOB).build();
+        assertTrue(JOHN.isSamePerson(editedJohn));
+
+        // different nric, same attributes -> returns false
+        editedJohn = new DonorBuilder(JOHN).withNric(VALID_NRIC_BOB).build();
+        assertFalse(JOHN.isSamePerson(editedJohn));
```

Comment:
This is the same as lines 30-31. Could you change the test for 29-30 such that it has different attributes and different nric also?